Lib/queue.py

```diff
@@ -198,6 +232,24 @@ def get_nowait(self):
         '''
         return self.get(block=False)
 
+    def shutdown(self, immediate=False):
+        '''Shut-down the queue, making queue gets and puts raise.
+
+        By default, gets will only raise once the queue is empty. Set
+        'immediate' to True to make gets raise immediately instead.
+
+        All blocked callers of put() will be unblocked, and also get()
+        and join() if 'immediate'. The ShutDown exception is raised.
+        '''
+        with self.mutex:
+            if immediate:
+                self.shutdown_state = _queue_shutdown_immediate
+                self.not_empty.notify_all()
+                self.all_tasks_done.notify_all()
```

Comment:
`self.unfinished_tasks` must be set to `0`

```suggestion
                # set self.unfinished_tasks to 0
                # to break the loop in 'self.join()'
                # when quits from `wait()`
                self.unfinished_tasks = 0
                self.all_tasks_done.notify_all()
```